### ISSUE TITLE
PHP8 Compatibility?

### ISSUE THREAD

**Issue by mikesoule** [pmjones/ext-request#50]
@pmjones I'm not able to install this with PHP 8 via Pecl or manually. Is this a known issue or possibly a bug in the distro I'm using (Alpine 3.13).

**Output from Pecl install:**
```
creating libtool
appending configuration tag "CXX" to libtool
configure: patching config.h.in
configure: creating ./config.status
config.status: creating config.h
running: make
/bin/sh /tmp/pear/temp/pear-build-defaultuserpaiPPL/request-2.0.0/libtool --mode=compile cc -I. -I/tmp/pear/temp/request -I/tmp/pear/temp/pear-build-defaultuserpaiPPL/request-2.0.0/include -I/tmp/pear/temp/pear-build-defaultuserpaiPPL/request-2.0.0/main -I/tmp/pear/temp/request -I/usr/include/php8 -I/usr/include/php8/main -I/usr/include/php8/TSRM -I/usr/include/php8/Zend -I/usr/include/php8/ext -I/usr/include/php8/ext/date/lib  -DHAVE_CONFIG_H  -g -O2    -c /tmp/pear/temp/request/php_request.c -o php_request.lo
mkdir .libs
 cc -I. -I/tmp/pear/temp/request -I/tmp/pear/temp/pear-build-defaultuserpaiPPL/request-2.0.0/include -I/tmp/pear/temp/pear-build-defaultuserpaiPPL/request-2.0.0/main -I/tmp/pear/temp/request -I/usr/include/php8 -I/usr/include/php8/main -I/usr/include/php8/TSRM -I/usr/include/php8/Zend -I/usr/include/php8/ext -I/usr/include/php8/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/pear/temp/request/php_request.c  -fPIC -DPIC -o .libs/php_request.o
(cd /tmp/pear/temp/request;  --no-generation-date -o request_parsers.c request_parsers.re)
/bin/sh: --no-generation-date: not found
make: *** [Makefile:209: /tmp/pear/temp/request/request_parsers.c] Error 127
ERROR: `make' failed
```

**Full install on Alpine:**
```bash
apk add autoconf automake gcc g++ libstdc++ linux-headers make musl php8 php8-dev php8-pear
pecl8 install request
```

Let me know if any more info will help.

**Comment by pmjones** [comment]
@jbboehr Does this look like a distro-based error to you, or does it sound like something with ext-request itself?

@mikesoule In the meantime, if you are eager to get moving with SapiRequest et al, you may wish to try the userland version at https://github.com/pmjones/php-request -- there are some necessary but minor differences from the extension, and of course it cannot be as fast since it is not in C.

**Comment by pmjones** [comment]
@mikesoule on looking at the output more closely, this bit `cd /tmp/pear/temp/request;  --no-generation-date` may be the problem. Looks like a flag is being specified, but there's no command before it.

**Comment by jbboehr** [comment]
They made some last-minute changes to the zend property macros. Fixed in #52
